### PR TITLE
Issue 1373: Add non-prod banner (RJM sandbox)

### LINKED ISSUE #1373: Indicate to user that they are using STAGING/development instead of Prod.
### Story

As a **user** of the registrar
I want be made perpetually aware that I am using STAGING or DEVELOPMENT
so that I know that I am testing and NOT affecting production data.

### Acceptance Criteria

- [ ] A banner or other mutually agreed indicator reminds me that I am in the a non production environment. This appears on every page while I am in that envirornment.
- [ ] Make sure the banner is active in manage and Django

### Additional Context

Data in staging and development may look **_similar or nearly identical_** to production data, so that it closely mimics the operational configuration. Therefore, we need to be careful to not confuse users, which might result in human errors when changing data.

While this was intended to just be for staging or development, there is no harm in having the banner show for developer sandboxes too by utilizing the is_production environment variable.

There is no design for the banner, thought typically this would be a red banner to make it obvious. 
### Issue Links

Relates to #1390

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #1373 

## Changes

- Alerts on the user-facing app and on django admin
- Check out the beautiful unit tests

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [x] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [x] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots

Screenshot 2024-02-02 at 3 14 18 PM
Screenshot 2024-02-02 at 3 14 25 PM

### CODE REVIEW

**Review by dave-kennedy-ecs — APPROVED** [review]
Pulled code, tested, reviewed code

File: src/registrar/assets/sass/_theme/_alerts.scss [hunk 1]
```diff
@@ -17,5 +17,8 @@
     .usa-alert__body::before {
       left: 1rem !important;
     }
-  } 
+  }
+  .usa-alert__body.margin-left-1 {
+    margin-left: 0.5rem!important;
```
Comment (zandercymatics) [inline src/registrar/assets/sass/_theme/_alerts.scss:22]:
```suggestion
    margin-left: 0.5rem !important;
```
(Nitpick)

File: src/registrar/templates/admin/base_site.html [hunk 2]
```diff
@@ -24,34 +24,57 @@
 {% block title %}{% if subtitle %}{{ subtitle }} | {% endif %}{{ title }} | {{ site_title|default:_('Django site admin') }}{% endblock %}
 
 {% block extrastyle %}{{ block.super }}
-<link rel="stylesheet" type="text/css" href="{% static "css/styles.css" %}" />
+<link rel="stylesheet" type="text/css" href="{% static 'css/styles.css' %}" />
```
Comment (zandercymatics) [inline src/registrar/templates/admin/base_site.html:27]: (Q) Why was this changed?

File: src/registrar/templates/admin/base_site.html [hunk 3]
```diff
@@ -24,34 +24,57 @@
...
+      {% include "includes/non-production-alert.html" %}
+    {% endwith %}
   {% endif %}
-      <a href="{% url 'admin:logout' %}" id="admin-logout-button">{% translate 'Log out' %}</a>
-  {% include "admin/color_theme_toggle.html" %}
- {% endblock %}
-{% block nav-global %}{% endblock %}
\ No newline at end of file
+
+  {# Djando update: this div will change to header #}
```
Comment (zandercymatics) [inline src/registrar/templates/admin/base_site.html:37]:
```suggestion
  {# Django update: this div will change to header #}
```
(The nitpick chronicles, pt. 2: revenge of the jando)

File: src/registrar/tests/test_environment_variables_effects.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    def tearDown(self):
+        super().tearDown()
+        self.user.delete()
+
+    @override_settings(IS_PRODUCTION=True)
+    def test_production_environment(self):
+        """No banner on prod."""
+        home_page = self.client.get("/")
+        self.assertNotContains(home_page, "You are not on production.")
```
Comment (zandercymatics) [inline src/registrar/tests/test_environment_variables_effects.py:25]: Nice! I had no idea you could do this. Love the tests

**Review by zandercymatics — APPROVED** [review]
LGTM. My only suggestion (though its not blocking) would be to add a bold "warning" to the message, as so
image

File: src/registrar/templates/includes/non-production-alert.html [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<div class="usa-alert usa-alert--warning usa-alert--slim margin-y-0 {% if add_class %}{{ add_class }}{% endif %}">
+    <div class="usa-alert__body {% if add_body_class %}{{ add_body_class }}{% endif %}">
+        You are not on production.
```
Comment (zandercymatics) [inline src/registrar/templates/includes/non-production-alert.html:3]:
```suggestion
        <b>WARNING:</b> You are not on production.
```
(Optional) I think this contrasts a bit better

File: src/registrar/tests/test_environment_variables_effects.py (hunk 4 above)
Comment (neilmb) [reply src/registrar/tests/test_environment_variables_effects.py:25]: +1 This is cool!

**Review by neilmb — APPROVED** [review]

File: src/registrar/templates/admin/base_site.html (hunk 2 above)
Comment (rachidatecs) [reply src/registrar/templates/admin/base_site.html:27]: My ide doesn't like double quotes nested inside double quotes (it's one of MY nitpicks)

File: src/registrar/assets/sass/_theme/_alerts.scss (hunk 1 above)
Comment (rachidatecs) [reply src/registrar/assets/sass/_theme/_alerts.scss:22]: (Nitpick) I'm team no space before !important

File: src/registrar/templates/includes/non-production-alert.html (hunk 5 above)
Comment (rachidatecs) [reply src/registrar/templates/includes/non-production-alert.html:3]: I like this. I'll commit and see what the designers think.

**Review by kristinacyin — CHANGES_REQUESTED** [review]
Could you change the message wording? From
> WARNING: You are not on production.

to

> ATTENTION: You are on a test site.

**Review by kristinacyin — CHANGES_REQUESTED** [review]
Also, sorry for adding additional criteria, but we (design team) that it'll be a good idea to have the banner stand out more. Not only do we want to change the message to "ATTENTION: You are on a test site." it'll be nice to also change the color to pink.

Gaby created a [Figma design](https://www.figma.com/file/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?type=design&node-id=2349%3A8658&mode=design&t=kQu4To3xO3MuL0P6-1) show how it should look (the pink version is on the very right)
A screenshot for reference: 
banner

Note that Gaby used the ["emergency" alert style](https://designsystem.digital.gov/components/alert/)

**Comment by rachidatecs** [comment]
Revised the copy and the color as instructed.

NOTE: I used the USWDS markup for emergency for this banner. Note that there are 2 colors for emergency, I overrode the base one. Do we want to override the dark one for the future?

Screenshot 2024-02-07 at 10 59 33 AM

NOTE 2: That pink is a new custom color. Should we consider using a color from the DHS palette?
 
Screenshot 2024-02-07 at 10 56 26 AM

**Review by kristinacyin — APPROVED** [review]
LGTM! 🥳 👏 

And don't worry about using a DHS color or using this color in the future - this is a very one-off situation that doesn't follow convention just because the only people who will see the banner are .gov team members.
